internal/namespaces/baremetal/v1alpha1/custom_server.go

```
@@ -21,19 +23,38 @@ func serverWaitCommand() *core.Command {
 	}
 
 	return &core.Command{
-		Short:     `Wait for a server to reach a stable state`,
+		Short:     `Wait for a server to reach a stable state (delivery and installation)`,
 		Long:      `Wait for a server to reach a stable state. This is similar to using --wait flag on other action commands, but without requiring a new action on the server.`,
 		Namespace: "baremetal",
 		Resource:  "server",
 		Verb:      "wait",
 		ArgsType:  reflect.TypeOf(serverWaitRequest{}),
 		Run: func(ctx context.Context, argsI interface{}) (i interface{}, err error) {
 			api := baremetal.NewAPI(core.ExtractClient(ctx))
-			return api.WaitForServer(&baremetal.WaitForServerRequest{
+			logger.Debugf("starting to wait for server to reach a stable delivery status")
+			server, err := api.WaitForServer(&baremetal.WaitForServerRequest{
 				ServerID: argsI.(*serverWaitRequest).ServerID,
 				Zone:     argsI.(*serverWaitRequest).Zone,
 				Timeout:  serverActionTimeout,
 			})
+			if err != nil {
+				return nil, err
+			}
+			if server.Status != baremetal.ServerStatusReady {
+				return nil, fmt.Errorf("server %s did not reach stable delivery status", server.ID)
```

add `server.Status`

---

Let's return a CLI error with a hint to help the user